### PR TITLE
Binary arithmetic for IPv4

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Follow-up fix for Replace domain IP types (IPv4, IPv6) with native https://github.com/ClickHouse/ClickHouse/pull/43221

### CODE REVIEW

File: src/Functions/FunctionBinaryArithmetic.h [hunk 1]
```diff
@@ -1209,6 +1213,17 @@ class FunctionBinaryArithmetic : public IFunction
             return arguments[0];
         }
 
+        /// Special case - one or both arguments are IPv4
+        if (isIPv4(arguments[0]) || isIPv4(arguments[1]))
```
Comment (novikd) [inline src/Functions/FunctionBinaryArithmetic.h:1217]: Needs a test, I am almost sure that it's always false.

**Comment by azat** [comment]
This looks like ready?
Just want to make sure that it will be included into 23.1 (since it is not far off).

File: src/Functions/FunctionBinaryArithmetic.h (hunk 1 above)
Comment (KochetovNicolai) [reply src/Functions/FunctionBinaryArithmetic.h:1217]: Actually, function isIPv4 is not connected with the template above :) It's
`inline bool isIPv4(const DataTypePtr & data_type) { return WhichDataType(data_type).isIPv4(); }` so should work aifak.
Test is added, I will merge.

**Comment by KochetovNicolai** [comment]
It's strange that `Bugfix validate check` failed. The query files to me with `Illegal types IPv4 and UInt64 of arguments of function modulo` on master

**Pull request merged**